backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/database/repos/AdministratorsRepository.kt

```diff
@@ -32,14 +35,22 @@ object AdministratorsRepository {
         }
     }
 
-    fun insert(project: String, email: String, password: String) {
+    fun insert(project: String, email: String, password: String, role: Role, regionId: Int? = null) {
         val projectId = ProjectEntity.find { Projects.project eq project }.firstOrNull()?.id
             ?: throw IllegalArgumentException("Project does not exist.")
 
+        if (role in setOf(Role.REGION_ADMIN, Role.REGION_MANAGER) && regionId == null) {
+            throw IllegalArgumentException("Role ${role.db_value} needs to have a region assigned.")
+        } else if (role in setOf(Role.PROJECT_ADMIN) && regionId != null) {
+            throw java.lang.IllegalArgumentException("Role ${role.db_value} cannot have a region assigned.")
+        }
```

Comment:
I think we are missing some cases here, what if the user has the role NO_RIGHTS?
I think this would pass this ifelseif.

Reply:
This is not an authorization check. I am just checking here whether the (role, region) combination makes sense. We could also enforce this using a DB constraint. what do you think?